### ISSUE TITLE
webpack not finding module on travis-ci

### ISSUE THREAD

**Issue by mitchallen** [travis-ci/travis-ci#8265]
I have a simple react component project. I can run webpack against the project successfully under the following:
* Mac OSX
* docker ubuntu trusty image
* docker travisci/ci-nodejs:packer-1494866191 image

But when I try to run the project under travis-ci on either Trusty or Precise I get this:

```
Version: webpack 3.5.5
Time: 143ms
   Asset       Size  Chunks             Chunk Names
index.js  815 bytes       0  [emitted]  main
   [0] ./src/index.js 316 bytes {0} [built] [failed] [1 error]
ERROR in ./src/index.js
Module parse failed: /home/travis/build/mitchallen/react-card-grey/src/index.js Unexpected token (25:2)
You may need an appropriate loader to handle this file type.
| 
| const CardGrey = ({ children }) => (
|   <div>
|    <MuiThemeProvider>
|       <Paper style={paperStyle} zDepth={5}>    
The command "webpack -p --config ./webpack.config.js" exited with 2.
```
I've hacked .travis.yml and webpack.config.js every which way trying to make it work, always with the same result.  Here is the latest attempt:

### .travis.yml:
```
dist: precise
language: node_js
node_js:
  - "stable"
skip_cleanup: true
before_install:
  - lsb_release -a
script:
  - webpack -p --config ./webpack.config.js
  # - npm run build
  # - npm test
after_success:
  - bash <(curl -s https://codecov.io/bash) -e TRAVIS_NODE_VERSION
```
### webpack.config.js
```
var path = require('path');
module.exports = {
  entry: './src/index.js',
  output: {
    path: path.resolve(__dirname, 'build'),
    filename: 'index.js',
    libraryTarget: 'commonjs2'
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        include: path.resolve(__dirname, 'src'),
        exclude: /(node_modules|bower_components|build)/,
        use: {
          loader: 'babel-loader',
          options: {
            presets: [ "env", "stage-0", "react" ],
            plugins: [
              "babel-plugin-transform-decorators-legacy",
              "babel-plugin-transform-react-jsx",
              "babel-plugin-transform-object-rest-spread"
            ]
          }
        }
      }
    ]
  },
  externals: {
    react: 'commonjs react'
  }
};
```
### Project:
[https://github.com/mitchallen/react-card-grey](https://github.com/mitchallen/react-card-grey)

**Comment by BanzaiMan** [comment]
Does this work on your machine?

It's complaining about the third column of this line: https://github.com/mitchallen/react-card-grey/blob/031fa66044be1310c9efa5905e45a7cffbff021c/src/index.js#L25, which is `<`.

I don't think this is valid JavaScript. Perhaps you meant to put lines 25 through 31 in `"`.

**Comment by BanzaiMan** [comment]
Or maybe something like this? https://stackoverflow.com/questions/36318529/webpack-unexpected-token

**Comment by mitchallen** [comment]
Yes, it works on my machine and as I mentioned at the top even a travis-ci docker image.  The code in the src is run through tools to convert it.  The problem seems to be that there is something about the image travis-ci.org uses for precise that is different from the travis docker image for node_js.

**Comment by BanzaiMan** [comment]
This is quite curious. Executing the steps defined in `.travis.yml` manually does pass the build, but running the compiled build script indeed result in a failure. To be sure, this seems like a bug *somewhere*, but it is not clear to me yet what.

**Comment by mitchallen** [comment]
I have one project where I set it up to work on both Travis CI and another CI platform. It passes on the other platform.  Though to be fair, webpack can be quirky.

**Comment by BanzaiMan** [comment]
Here's the compiled script: https://gist.github.com/BanzaiMan/b6ab0b8bd3ba2ef0e36cc0250491529b#file-build-sh

**Comment by raRaRa** [comment]
@mitchallen My co-worker ran into the same issue, who also happens to exclude the build folder in the exclude line:
`exclude: /(node_modules|bower_components|build)/,`

My theory is that Travis is excluding everything since Travis puts everything under: /home/travis/**build**/...

Try removing build from the exclude line and check if that helps.

**Comment by Salinn** [comment]
Running into the same issue with running webpack and excluding those files

**Comment by vinodronold** [comment]
I too ran into same issue.. 
Updated the Webpack config to output the bundle to "lib" and excluded "lib" instead of "build" folder. 
Thank you @raRaRa

**Comment by BanzaiMan** [comment]
Can this be closed?

**Comment by Salinn** [comment]
@BanzaiMan still having an issue, just tried again

**Comment by raRaRa** [comment]
@Salinn Did you remove build from the excluded line?

**Comment by Salinn** [comment]
@raRaRa that does fix my issue
